Process 1.6.13 changed observable behaviour

Reproducer is https://github.com/angerman/repro-process-behaviour-change-1.6.13

Will produce the following output:
```sh
$ make all CABAL=$(which cabal) GHC=$(which ghc)
Building against process < 1.6.13 ...
Running against process < 1.6.13; with an empty PATH, this should result in a git not found message
<no git executable found>
Building against process >= 1.6.13 ...
Running against process >= 1.6.13; this now throws an exception instead.
process-crash: git: readCreateProcessWithExitCode: posix_spawnp: failed (Undefined error: 0)
make: *** [Makefile:9: all] Error 1
```

We are now throwing an exception where we didn't before.

--- a/cbits/posix/posix_spawn.c
+++ b/cbits/posix/posix_spawn.c
@@ -201,6 +201,7 @@ do_spawn_posix (char *const args[],
 
     r = posix_spawnp(&pid, args[0], &fa, &sa, args, environment ? environment : environ);
     if (r != 0) {
+        errno = r; // posix_spawn doesn't necessarily set errno; see #227.
         *failed_doing = "posix_spawnp";
         goto fail;
     } else {

